lib/improver/tests/utilities/test_DifferenceBetweenAdjacentGridSquares.py

```diff
@@ -130,6 +130,17 @@ def test_othercoords(self):
         self.assertEqual(result.coord(axis='x'), proj_x_coord)
         self.assertEqual(result.coord('time'), time_coord)
 
+    def test_gradient(self):
+        """Test that the correct metadata is set if the desired output is a
+        gradient not a difference (except name)"""
+        plugin = DifferenceBetweenAdjacentGridSquares(gradient=True)
+        diff_array = np.array([[1, 2, 3],
+                               [3, 6, 9]])
+        result = plugin.create_difference_cube(
+            self.cube, "projection_y_coordinate", diff_array)
+        self.assertNotIn("form_of_difference", result.attributes)
```

Comment:
We may need some kind of metadata for direction. See comments on relevant section of code.

Reply:
True.  Can I suggest just adding a TODO here, since we've been talking about the appropriate metadata conventions on another ticket, and given that's probably imminent we could add the right attributes then?